# Patch for the intel compiler.
# It changes a header file of the included Qhull library, as suggested
# by one of the CASM developers.
# It also changes the SConstruct script to use the PATH from the environment
# and to set compiler options for Intel rather than GNU.
# Developed by Kurt Lust, UAntwerpen.
--- CASMcode-0.2.0-orig/include/casm/external/qhull/libqhull_r/qhull_ra.h	2016-08-14 21:21:43.000000000 +0200
+++ CASMcode-0.2.0/include/casm/external/qhull/libqhull_r/qhull_ra.h	2017-03-22 16:53:02.010862000 +0100
@@ -102,13 +102,7 @@
 #elif defined(__MWERKS__) && defined(__INTEL__)
 #   define QHULL_OS_WIN
 #endif
-#if defined(__cplusplus) && defined(__INTEL_COMPILER) && !defined(QHULL_OS_WIN)
-template <typename T>
-inline void qhullUnused(typename T &x) { (void)x; }
-#  define QHULL_UNUSED(x) qhullUnused(x);
-#else
 #  define QHULL_UNUSED(x) (void)x;
-#endif
 
 /***** -libqhull_r.c prototypes (alphabetical after qhull) ********************/
 
--- CASMcode-0.2.0-orig/SConstruct	2016-08-14 21:21:43.000000000 +0200
+++ CASMcode-0.2.0/SConstruct	2017-03-23 14:42:46.633822059 +0100
@@ -209,9 +209,6 @@
   ccflags = []
   cxxflags = []
 
-  #ccflags.append('-Wall')
-  ccflags.append('-Wno-unused-parameter')
-
   _debug_level = debug_level()
   
   if _debug_level == '0':
@@ -227,10 +224,7 @@
   cxxflags.append(_opt)
 
   # C++ only
-  #cxxflags = []
-  cxxflags.append('--std=c++11')
-  cxxflags.append('-Wno-deprecated-register')
-  cxxflags.append('-Wno-deprecated-declarations')
+  cxxflags.append('-std=c++11')
   cxxflags.append('-DEIGEN_DEFAULT_DENSE_INDEX_TYPE=long')
 
   if boost_no_cxx11_scoped_enums():
@@ -267,6 +261,7 @@
 
 # where the headers are
 env.Append(INCDIR = join(os.getcwd(), 'include'))
+env['ENV']['CPATH'] = os.environ['CPATH']
 
 # where test binaries should go
 env.Append(UNIT_TEST_BINDIR = join(os.getcwd(), 'tests', 'unit', 'bin'))
@@ -313,7 +308,7 @@
 env['ENV']['TERM'] = os.environ['TERM']
 
 # set testing environment (for running tests)
-env['ENV']['PATH'] = env['BINDIR'] + ":" + env['ENV']['PATH']
+env['ENV']['PATH'] = env['BINDIR'] + ":" + os.environ['PATH']
 
 # set LD_LIBRARY_PATH or DYLD_FALLBACK_LIBRARY_PATH (for running configuration tests)
 for x in ['LD_LIBRARY_PATH', 'DYLD_LIBRARY_FALLBACK_PATH']:
